demos/jobs/jobs_demo_mosquitto/jobs_demo_mosquitto.c

```diff
@@ -0,0 +1,1264 @@
+             "Client device1 sending CONNECT\n"
+             "Client device1 received CONNACK\n"
+             "Client device1 sending SUBSCRIBE (Mid: 1, Topic: $aws/things/device1/jobs/start-next/accepted, QoS: 1)\n"
+             "Client device1 received SUBACK\n"
+             "[...]\n"
+             "starting job id: t12\n"
+             "sending first update\n" );
+    fprintf( stderr,
+             "\nIf the output does not show a successful connection, check in the AWS console\n"
+             "that the client certificate is associated with the target thing and is activated.\n"
+             "Also check that the Amazon Root CA certificates are in your system's trust store.\n"
+             "Note, you can provide a CA certificate file directly as a command-line argument.\n" );
+    fprintf( stderr,
+             "\nThis demonstration exits on most error conditions.  One way to retry while avoiding\n"
+             "throttling due to excessive reconnects is to periodically relaunch from cron(8).\n"
+             "Given a shell script wrapper with the necessary arguments called download, the following\n"
+             "line in /etc/crontab would start the downloader unless it is already running.\n"
+             "This tries every 3 minutes, with an additional random delay up to 2 minutes.\n\n"
+             "*/3 * * * *  root  exec 9> /tmp/lock && flock -n 9 && sleep $((RANDOM %% 120)) && download\n"
+             );
+    fprintf( stderr,
+             "\nusage: %s "
+             "-n name -h host [-p port] {--cafile file | --capath dir} --cert file --key file\n"
+             "\n"
+             "-n : thing name\n"
+             "-h : mqtt host to connect to.\n"
+             "-p : network port to connect to. Defaults to %d.\n",
+             programName, DEFAULT_MQTT_PORT );
+    fprintf( stderr,
+             "--cafile : path to a file containing trusted CA certificates to enable encrypted\n"
+             "           certificate based communication.\n"
+             "--capath : path to a directory containing trusted CA certificates to enable encrypted\n"
+             "           communication.  Defaults to %s.\n"
+             "--cert : client certificate for authentication.\n"
+             "--key : client private key for authentication.\n\n",
+             DEFAULT_CA_DIRECTORY );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief the several states of execution
+ */
+typedef enum
+{
+    None = 0, /* no current job */
+    Ready,    /* job document received and parsed */
+    Running,  /* download in progress */
+    Cancel,   /* cancel due to failed update */
+} runStatus_t;
+
+/**
+ * @brief all runtime parameters and state
+ */
+typedef struct
+{
+    /* thing name */
+    char * name;
+    size_t nameLength;
+    /* connection parameters */
+    char * host;
+    uint16_t port;
+    char * cafile;
+    char * capath;
+    char * cert;
+    char * key;
```

Comment:
It is not clear whether they contain the contents of the certificate or the file path, so I guess update or document cert and path to be labeled as a file as well.

Reply:
changed